### PR TITLE
[Core: TimePoint] Add type declarations to function signatures

### PR DESCRIPTION
**Pull request by johnsaigle**
See #3878 for a better description.

### CODE REVIEW

File: php/libraries/TimePoint.class.inc [hunk 1]
```diff
@@ -726,22 +737,6 @@ class TimePoint
         $this->select($this->getSessionID());
     }
 
-    /**
-     * THIS FUNCTION DOES NOTHING. IT SHOULD BE REMOVED.
```
Comment (johnsaigle) [inline php/libraries/TimePoint.class.inc]: Done! 😄

File: php/libraries/TimePoint.class.inc [hunk 2]
```diff
@@ -335,12 +346,12 @@ class TimePoint
...
      *
-     * @param string $var   The variable to set for this timepoint.
-     * @param string $value The value to set $var to.
+     * @param string      $var   The variable to set for this timepoint.
+     * @param string|null $value The value to set $var to.
      *
      * @return void (side-effect modifies session table.)
      */
-    function setData($var, $value = null)
+    function setData($var, ?string $value = null): void
```
Comment (johnsaigle) [inline php/libraries/TimePoint.class.inc]: See #4050 for more information and why this cannot yet have a type declaration

File: php/libraries/TimePoint.class.inc [hunk 3]
```diff
@@ -185,16 +188,20 @@ class TimePoint
...
-     * @param integer $candID         6 digit CandID having a timepoint created.
-     * @param integer $SubprojectID   The subprojectID of the new timepoint
-     * @param string  $nextVisitLabel The visit label of the timepoint being created
-     * @param string  $psc            The centre of the timepoint
+     * @param integer     $candID         6 digit CandID having a timepoint created.
+     * @param integer     $SubprojectID   The subprojectID of the new timepoint
+     * @param string|null $nextVisitLabel The visit label of the timepoint
+     *                                              being created
+     * @param string|null $psc            The centre of the timepoint
      *
```
Comment (PapillonMcGill) [inline php/libraries/TimePoint.class.inc]: Should we add @throw database exception?

File: php/libraries/TimePoint.class.inc [hunk 4]
```diff
@@ -450,37 +485,37 @@ class TimePoint
...
     }
 
     /**
      * Returns the date of screening for this session.
      *
-     * @return string of form "YYYY-MM-DD" (or null)
+     * @return string|null of form "YYYY-MM-DD" (or null)
      */
-    function getDateOfScreening()
+    function getDateOfScreening(): ?string
```
Comment (PapillonMcGill) [inline php/libraries/TimePoint.class.inc]: require PHP 7.1

File: php/libraries/TimePoint.class.inc (hunk 3 above)
Comment (PapillonMcGill) [reply php/libraries/TimePoint.class.inc]: waiting Travis

**Comment by johnsaigle** [comment]
Blocked by #4066

**Pull request merged**